hooks/environment

```diff
@@ -10,7 +10,13 @@ function set_env() {
 }
 
 echo "--- :house_with_garden: Setting up environment"
-if [[ -v BUILDKITE_PLUGIN_SMOOTH_CHECKOUT_BUILD_CHECKOUT_PATH ]]; then
+
+if [[ -v BUILDKITE_PLUGIN_SMOOTH_CHECKOUT_INTERPOLATE_CHECKOUT_PATH ]]; then
+  INTERPOLATED_CHECKOUT_PATH=$(eval echo "$BUILDKITE_PLUGIN_SMOOTH_CHECKOUT_BUILD_CHECKOUT_PATH")
+  set_env BUILDKITE_BUILD_CHECKOUT_PATH "$INTERPOLATED_CHECKOUT_PATH"
+  # WORKSPACE is deprecated and preserved for backward compatibility.
+  set_env WORKSPACE "${BUILDKITE_BUILD_CHECKOUT_PATH}"
```

Comment:
I think setting `WORKSPACE` env is common for both the `if` and `elif` cases here. So maybe we do it in a common place after line 23.